packages/babel-core/src/transformation/file/options/option-manager.js

```diff
@@ -261,6 +261,18 @@ export default class OptionManager {
       try {
         if (typeof val === "string") {
           presetLoc = resolve(`babel-preset-${val}`, dirname) || resolve(val, dirname);
```

Comment:
`babel-preset-${val}` will never be valid if val contains an @, but I think doing a check if it contains an @  and then doing different things makes it even more complex and works anyway.